pkg/generate/app/cmd/newapp.go

```diff
@@ -386,6 +425,19 @@ func (c *AppConfig) Run(out io.Writer) (*AppResult, error) {
 
 	objects = app.AddServices(objects)
 
+	templates, err := c.buildTemplates(components, app.Environment(parameters))
+	if err != nil {
+		return nil, err
+	}
+	for _, t := range templates {
+		templateResolver := c.templateResolver.(app.TemplateResolver)
+		items, err := t.Process(templateResolver.TemplateConfigsNamespacer, templateResolver.Namespaces[0])
```

Comment:
:) yeah, it's ok to get info from the resolver but it's probably better to keep it distinct

> On Apr 19, 2015, at 4:52 PM, Fabiano Franz <notifications@github.com> wrote:
> 
> In pkg/generate/app/cmd/newapp.go:
> 
> > @@ -386,6 +425,19 @@ func (c *AppConfig) Run(out io.Writer) (*AppResult, error) {
> >  
> >  	objects = app.AddServices(objects)
> >  
> > +	templates, err := c.buildTemplates(components, app.Environment(parameters))
> > +	if err != nil {
> > +		return nil, err
> > +	}
> > +	for _, t := range templates {
> > +		templateResolver := c.templateResolver.(app.TemplateResolver)
> > +		items, err := t.Process(templateResolver.TemplateConfigsNamespacer, templateResolver.Namespaces[0])
> You got the "found breaking encapsulation piece" trophy. I knew you could do it people, was just waiting who was going to find it first! :)
> 
> —
> Reply to this email directly or view it on GitHub.
>